packages/ag-charts-community/src/chart/options/themeOptions.ts

```diff
@@ -48,30 +39,24 @@ export interface AgBaseChartTheme<T> extends AgBaseChartThemeOptions<T> {
     baseTheme?: AgChartThemeName;
 }
 
-export interface AgBaseChartThemeOverrides<T> {
-    /** Specifies defaults for all cartesian charts (used for bar, column, histogram, line, scatter and area series) */
-    cartesian?: AgBaseCartesianThemeOptions<AgCartesianSeriesTheme> & T;
-    /** Specifies defaults for bar charts. */
-    bar?: AgBaseCartesianThemeOptions<AgBarSeriesOptions> & T;
-    /** Specifies defaults for line charts. */
-    line?: AgBaseCartesianThemeOptions<AgLineSeriesOptions> & T;
-    /** Specifies defaults for area charts. */
-    area?: AgBaseCartesianThemeOptions<AgAreaSeriesOptions> & T;
-    /** Specifies defaults for scatter/bubble charts. */
-    scatter?: AgBaseCartesianThemeOptions<AgScatterSeriesOptions> & T;
-    /** Specifies defaults for histogram charts. */
-    histogram?: AgBaseCartesianThemeOptions<AgHistogramSeriesOptions> & T;
-
-    /** Specifies defaults for all polar charts (used for pie series) */
-    polar?: AgBasePolarThemeOptions<AgPolarSeriesTheme> & T;
-    /** Specifies defaults for pie/doughnut charts. */
-    pie?: AgBasePolarThemeOptions<AgPieSeriesOptions> & T;
-
-    /** Specifies defaults for all hierarchy charts (used for treemap series) */
-    hierarchy?: AgBaseHierarchyThemeOptions<AgHierarchySeriesTheme> & T;
-    /** Specifies defaults for all treemap charts. */
-    treemap?: AgBaseHierarchyThemeOptions<AgHierarchySeriesOptions> & T;
-
-    /** Specifies defaults for all chart types. Be careful to only use properties that apply to all chart types here. For example, don't specify `navigator` configuration here as navigators are only available in cartesian charts. */
-    common?: any;
-}
+type SeriesThemeSkipOptions = 'id' | 'xKey' | 'yKey' | 'type';
+type SeriesThemeOptions<Series, Type> = Omit<Partial<Extract<Series, { type?: Type }>>, SeriesThemeSkipOptions>;
+
+export type AgBaseChartThemeOverrides<T> = {
+    [K in NonNullable<AgCartesianSeriesOptions['type']>]?: AgBaseCartesianThemeOptions<
```

Comment:
what if there are overlaps with the  series options?

Reply:
Discussed offline - there's no overlap.